tests/Integration/IntegrationTest.php

```diff
@@ -0,0 +1,206 @@
+            );
+
+            $outputBuffer = new BufferedOutput();
+
+            $command->run(
+                $input,
+                $outputBuffer,
+            );
+            if (!file_exists($expectedPath . '/logs/error.log')) {
+                self::assertFileDoesNotExist($outputPath . '/logs/error.log');
+            }
+
+            if (!file_exists($expectedPath . '/logs/warning.log')) {
+                self::assertFileDoesNotExist($outputPath . '/logs/warning.log');
+            }
+
+            foreach ($compareFiles as $compareFile) {
+                $outputFile = str_replace($expectedPath, $outputPath, $compareFile);
+                if (str_ends_with($compareFile, '.log')) {
+                    self::assertFileContainsLines($compareFile, $outputFile);
+                } else {
+                    self::assertFileEqualsTrimmed($compareFile, $outputFile, 'Expected file path: ' . $compareFile);
+                }
+            }
+        } catch (ExpectationFailedException $e) {
+            if ($skip) {
+                self::markTestIncomplete(file_get_contents($inputPath . '/skip') ?: '');
+            }
+
+            throw $e;
+        }
+
+        self::assertFalse($skip, 'Test passes while marked as SKIP.');
+    }
+
+    /**
+     * Asserts that each line of the expected file is contained in actual
+     *
+     * @throws ExpectationFailedException
+     */
+    public static function assertFileContainsLines(string $expected, string $actual): void
+    {
+        self::assertFileExists($expected);
+        self::assertFileExists($actual);
+
+        $lines = explode("\n", file_get_contents($expected));
+        $actualContent =  file_get_contents($actual);
+        foreach ($lines as $line) {
+            self::assertStringContainsString($line, $actualContent, 'File "' . $actual . '" does not contain "' . $line . '"');
+        }
+    }
+
+    /**
+     * Asserts that the contents of one file is equal to the contents of another
+     * file. It ignores empty lines and whitespace at the start and end of each line
+     *
+     * @throws ExpectationFailedException
+     */
+    public static function assertFileEqualsTrimmed(string $expected, string $actual, string $message = ''): void
+    {
+        self::assertFileExists($expected, $message);
+        self::assertFileExists($actual, $message);
+
+        self::assertEquals(self::getTrimmedFileContent($expected), self::getTrimmedFileContent($actual), $message);
+    }
+
+    public static function getTrimmedFileContent(string $file): string
+    {
+        $contentArray = explode("\n", file_get_contents($file));
+        array_walk($contentArray, static function (&$value): void {
+            $value = trim($value);
+        });
+        $contentArray = array_filter($contentArray, static function ($value) {
+            return $value !== '';
+        });
+
+        return implode("\n", $contentArray);
+    }
+
+    /** @return mixed[] */
```

Comment:
I don't think this is useful. Do we really want to have this on tests, then we should annotate the proper return type. Also below.

Reply:
changed in all locations